### ISSUE TITLE
ラベルが見つかりませんと警告文が出てきます。

### ISSUE THREAD

**Issue by akatsuryo** [ultralytics/ultralytics#14347]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

#自己紹介#
私は日本の学生です。
詳細に情報を伝えたいので日本語を使用しています。ご了承ください。
##今回の目的##
自作セグメントモデルを完成させる。
##エラー文##
New https://pypi.org/project/ultralytics/8.2.53 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.52 🚀 Python-3.12.4 torch-2.3.1+cpu CPU (Intel Core(TM) i7-6800K 3.40GHz)
engine\trainer: task=segment, mode=train, model=yolov8n-seg.pt, data=data.yml, epochs=10, time=None, patience=100, batch=8, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=4, project=None, name=train4, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\segment\train4
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]
 22        [15, 18, 21]  1   1004275  ultralytics.nn.modules.head.Segment          [1, 32, 64, [64, 128, 256]]
YOLOv8n-seg summary: 261 layers, 3263811 parameters, 3263795 gradients, 12.1 GFLOPs

Transferred 381/417 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
train: Scanning C:\Users\myacc\Downloads\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data\train\labels.cache
WARNING ⚠️ No labels found in C:\Users\myacc\Downloads\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data\train\labels.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
val: Scanning C:\Users\myacc\Downloads\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data\val\labels.cache...
WARNING ⚠️ No labels found in C:\Users\myacc\Downloads\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data\val\labels.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
Plotting labels to runs\segment\train4\labels.jpg...
zero-size array to reduction operation maximum which has no identity
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 66 weight(decay=0.0), 77 weight(decay=0.0005), 76 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs\segment\train4
Starting training for 10 epochs...
Closing dataloader mosaic

##フォルダの階層##
[C:\Users\myacc\Downloads\全授業資料\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data]
[\yolo_data\train\images]
[\yolo_data\train\labels]
[\yolo_data\val\images]
[\yolo_data\val\labels]
##pythonファイルのパス##
[\yolo_data\Yolo_v8_seg.py]
##YAMLファイルのパス##
[\yolo_data\data.yml]
###pythonファイルの中身###
from ultralytics import YOLO

model = YOLO("yolov8n-seg.pt")

model.train(data="data.yml", epochs=10, batch=8, workers=4)
###YAMLファイルの中身###
path: C:\Users\myacc\Downloads\全授業資料\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data
train: train/images
val: val/images

names:
  0: Road
##labelのファイル名##
1～n番までのテキストファイル
例：1.txt 2.txt ・・・n.txt
##imageのファイル名##
1～n番までのjpgファイル
例：1.jpg 2.jpg ・・・n.jpg

###labelの中身###
0 0.4062 0.4375 0.4102 0.4414 0.4062 0.4453 0.4062 0.4531 0.4141 0.4609 0.4102 0.4648・・・

※0はRoadのこと。[0.4062 0.4375 0.4102 0.4414]はx1 y1 x2 y2 のこと。座標はポリゴンの座標値を相対座標にしたもの。

**Comment by glenn-jocher** [comment]
@akatsuryo こんにちは！ご質問ありがとうございます。

エラーメッセージから判断すると、ラベルが見つからないため、トレーニングが正しく行われない可能性があります。以下の点を確認してみてください：

1. **ラベルファイルの確認**:
   - 各画像に対応するラベルファイルが存在するか確認してください。ラベルファイルは画像ファイルと同じ名前で、拡張子が`.txt`である必要があります。
   - ラベルファイルの中身が正しいフォーマットで記述されているか確認してください。例えば、以下のような形式です：
     ```
     0 0.4062 0.4375 0.4102 0.4414 0.4062 0.4453 0.4062 0.4531 0.4141 0.4609 0.4102 0.4648
     ```

2. **YAMLファイルの確認**:
   - `data.yml`ファイルのパスが正しいか確認してください。特に、`path`フィールドが正しいディレクトリを指しているか確認してください。
   - `train`と`val`フィールドが正しいディレクトリを指しているか確認してください。

3. **最新バージョンの使用**:
   - 最新バージョンのUltralyticsパッケージを使用しているか確認してください。以下のコマンドでアップデートできます：
     ```bash
     pip install -U ultralytics
     ```

4. **再現可能な例の提供**:
   - 問題が解決しない場合、再現可能な最小限のコード例を提供していただけると助かります。こちらのリンクを参考にしてください：https://docs.ultralytics.com/help/minimum_reproducible_example

以下は、ラベルファイルとYAMLファイルの例です：

### ラベルファイルの例
```
0 0.4062 0.4375 0.4102 0.4414 0.4062 0.4453 0.4062 0.4531 0.4141 0.4609 0.4102 0.4648
```

### YAMLファイルの例
```yaml
path: C:\Users\myacc\Downloads\全授業資料\データサイエンスプロジェクト3\Yolov8_test\dataset\yolo_data
train: train/images
val: val/images

names:
  0: Road
```

これらの確認を行った後でも問題が解決しない場合は、再度ご連絡ください。ご協力ありがとうございます！🚀

**Comment by akatsuryo** [comment]
沢山の返信ありがとうございました。原因はファイル名にありました。ご協力ありがとうございました。

**Comment by pderrenger** [comment]
@akatsuryo ファイル名が原因だったとのこと、解決できて良かったです。何か他にご質問や問題がありましたら、いつでもお知らせください。引き続き頑張ってください。
